.github/workflows/test-pr.yml

```diff
@@ -41,10 +41,83 @@ jobs:
       - name: 'Set up Docker'
         uses: ./.github/actions/with-docker
         with:
-          container-name: kwasm-ci-${{ github.sha }}
+          container-name: kwasm-ci-pykwasm-${{ github.sha }}
       - name: 'Build and run integration tests'
-        run: docker exec -u user kwasm-ci-${GITHUB_SHA} make -C pykwasm cov-integration
+        run: docker exec -u user kwasm-ci-pykwasm-${GITHUB_SHA} make -C pykwasm cov-integration
       - name: 'Tear down Docker'
         if: always()
         run: |
-          docker stop --time=0 kwasm-ci-${GITHUB_SHA}
+          docker stop --time=0 kwasm-ci-pykwasm-${GITHUB_SHA}
+
+  parse-tests:
+    needs: [pykwasm-unit-tests, pykwasm-integration-tests]
+    name: 'Parser Tests'
+    runs-on: [self-hosted, linux, normal]
+    timeout-minutes: 15
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: 'Set up Docker'
+        uses: ./.github/actions/with-docker
+        with:
+          container-name: kwasm-ci-parse-${{ github.sha }}
+      - name: 'Build LLVM Backend and pykwasm'
+        run: docker exec -u user kwasm-ci-parse-${GITHUB_SHA} make -j2 build-llvm pykwasm-poetry-install
+      - name: 'Binary parse'
+        run: docker exec -u user kwasm-ci-parse-${GITHUB_SHA} make -j6 test-binary-parser
+      - name: 'Conformance parse'
+        run: docker exec -u user kwasm-ci-parse-${GITHUB_SHA} make -j6 test-conformance-parse
+      - name: 'Tear down Docker'
+        if: always()
+        run: |
+          docker stop --time=0 kwasm-ci-parse-${GITHUB_SHA}
+
+  conformance-tests:
+    needs: [pykwasm-unit-tests, pykwasm-integration-tests]
+    name: 'Conformance Tests'
+    runs-on: [self-hosted, linux, normal]
+    timeout-minutes: 15
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: 'Set up Docker'
+        uses: ./.github/actions/with-docker
+        with:
+          container-name: kwasm-ci-conformance-${{ github.sha }}
+      - name: 'Build LLVM Backend'
+        run: docker exec -u user kwasm-ci-conformance-${GITHUB_SHA} make -j2 build-llvm
+      - name: 'Simple tests'
+        run: docker exec -u user kwasm-ci-conformance-${GITHUB_SHA} make -j6 test-simple
```

Comment:
Also appears to be missing TEST_CONCRETE
`make TEST_CONCRETE_BACKEND=llvm test-simple -j4`